ra/ra.go

```diff
@@ -958,27 +969,59 @@ func (ra *RegistrationAuthorityImpl) FinalizeOrder(ctx context.Context, req *rap
 		}
 	}
 
+	err = processOrder(ctx, ra, order, csrOb)
+	if err != nil {
+		return nil, err
+	}
+
+	// Note how many names were in this finalized certificate order.
+	ra.namesPerCert.With(
+		prometheus.Labels{"type": "issued"},
+	).Observe(float64(len(order.Names)))
+
+	// Update the order status locally since the SA doesn't return the updated
+	// order itself after setting the status
+	order.Status = string(core.StatusValid)
+	return order, nil
+}
+
+// Process the order through the StorageAuthority and issue a certificate.
+// If any error happens during the process, fail the order and return the error.
+func processOrder(
+	ctx context.Context,
+	ra *RegistrationAuthorityImpl,
+	order *corepb.Order,
+	csrOb *x509.CertificateRequest) error {
+	// Create a goroutine to handle marking the order as failed.
+	// We attach a separate context to prevent that the original context expiration
+	// prevents the order to be marked as failed.
+	problems := make(chan *probs.ProblemDetails)
+	failCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
I don't have any idea what a suitable value for this timeout would be.
I noticed there's an `orderLifetime` field in regiatration authority, but I am not sure if it's a good idea to use it for this context's expiration deadline?

Reply:
This is a good question!

`orderLifetime` is definitely not what we want: that's now long an Order is allowed to sit in our database before we consider it too old and age it out. It's on the order of many days, not a few seconds like we want here.

I think a timeout of 1 second is probably appropriate for the failCtx. We expect all of our API methods to be quite fast, and this is just a single database write. If it takes more than 1 second to write one column to one row of the database, something has gone very wrong.